src/test/ui/const-generics/defaults/right-order.rs

```diff
@@ -0,0 +1,11 @@
+// run-pass
+// Verifies that having generic parameters after constants is permitted.
+
+#![feature(const_generics)]
+#![allow(incomplete_features)]
+
+struct A<const N: usize, T=u32>(T);
```

Comment:
Imo it's probably better to unify this test with `src/test/ui/const-generics/defaults/needs-feature.rs` and use revisions here.